### PR TITLE
EKS k8s audit logs plugin

### PR DESCRIPTION
**Pull request by Issif**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area plugins

> /area registry

> /area build

> /area documentation

**What this PR does / why we need it**:

This PR adds a new plugin for k8s audit logs for EKS from Cloudwatch, it is based on the `k8saudit` plugin for field extraction and on shared libs for AWS services, see #131 

Example:
```
root@falco:/etc/falco/plugins/k8sauditeks# falco -c falco.yaml -r rules/k8s_audit_rules.yaml 
15:12:43.826983000: Warning K8s Operation performed by user not in allowed list of users (user=eks:addon-manager target=coredns/serviceaccounts verb=patch uri=/api/v1/namespaces/kube-system/serviceaccounts/coredns?fieldManager=eks&force=true&timeout=10s resp=200)
15:12:43.829125000: Informational K8s Serviceaccount Created (user=system:kube-controller-manager serviceaccount=resourcequota-controller ns=kube-system resp=201 decision=allow reason=RBAC: allowed by ClusterRoleBinding "system:kube-controller-manager" of ClusterRole "system:kube-controller-manager" to User "system:kube-controller-manager")
```

**Which issue(s) this PR fixes**:

This plugin depends of #131 to test it, you need to add `replace` in `go.mod`, and as I forked my branch for the shared libs, this current PR includes the changes for it too. After the merge of #131 it will be easier to review

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: plugins/k8sauditeks/Makefile
Comment (jasondellaluce) [inline plugins/k8sauditeks/Makefile:36]: This is a copy-paste from k8saudit I guess! I think we should not include this for now to reduce entropy. This was put there just to provide a barebones benchmark attempt, but we need a better testing strategy in the general case.

File: plugins/k8sauditeks/Makefile
Comment (jasondellaluce) [inline plugins/k8sauditeks/Makefile:22]: I believe a better name for this pluginnwould be `eks-k8saudit`. It's more readable and it sorts of hints that this is a "flavor" of k8saudit. It's also in line with the current sample docs where this was first envisioned and mentioned: https://falco.org/docs/plugins/#plugin-event-sources-and-interoperability

File: plugins/k8sauditeks/README.md
Comment (jasondellaluce) [inline plugins/k8sauditeks/README.md:40]: Let's put this as true by default. It's a huge performance gain. See: https://github.com/falcosecurity/plugin-sdk-go/issues/62

File: plugins/k8sauditeks/README.md
Comment (jasondellaluce) [inline plugins/k8sauditeks/README.md:35]: I see `region`, `profile`, and `cluster_name` as good candidates for open parameters instead than init configuration. Do you need these elsewhere out of the generation of events? If not, these are good open parameters for the event source.

Since that is usually a string, you should propose some URI-like encoding form.

File: plugins/k8sauditeks/README.md
Comment (jasondellaluce) [inline plugins/k8sauditeks/README.md:61]: Replicating the whole ruleset is a bad choice here. This plugin should not ship any ruleset, and be compatible with the `k8saudit` one. I see why you couldn't do it, and I opened an issue for this problem specifically: https://github.com/falcosecurity/falco/issues/2150

So I think our choice here would be to:
- Block this PR until https://github.com/falcosecurity/falco/issues/2150 is merged
- Remove the `k8saudit` plugin dependency on the ruleset and keep only `json`, so that this plugin gets "implicitly" accepted too

File: plugins/k8sauditeks/falco.yaml
Comment (jasondellaluce) [inline plugins/k8sauditeks/falco.yaml:4]: Leftovers? I see no point in having a sample falco.yaml here anyway since we have all this info in the README

File: plugins/k8sauditeks/go.mod
Comment (jasondellaluce) [inline plugins/k8sauditeks/go.mod:1]: This is not consitent with the name of the directory. Consider my naming suggestion above here too.

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:43]: Same as before: Region, Profile, and Cluster name are good open parameters candidates.

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:78]: I think region should not be optional here. Another point in favor of making it an open param.

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:114]: Maybe `kube-apiserver-audit` should be configurable too.

File: plugins/k8sauditeks/rules/k8s_audit_rules.yaml
Comment (jasondellaluce) [inline plugins/k8sauditeks/rules/k8s_audit_rules.yaml:1]: This file is a replica of the k8saudit one and I'm against having two copies, see comment above.

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:128]: You should use the `k8saudit.ParseAuditEventsPayload` method here before pushing to the channel. That would allow for content checking and timestamp setting. See how the `parseAuditEventsAndPush` private method is implemented (and how it logs errors).

**Review by jasondellaluce — COMMENTED** [review]
Great work! I left a first round of comments

File: plugins/k8sauditeks/Makefile
Comment (Issif) [reply plugins/k8sauditeks/Makefile:36]: I'll clean that.

File: plugins/k8sauditeks/Makefile
Comment (Issif) [reply plugins/k8sauditeks/Makefile:22]: I prefer `k8saudit-eks`, it allows to have all `k8saudit-X` plugins listed side by side in the repository, wdyt?

File: plugins/k8sauditeks/README.md
Comment (Issif) [reply plugins/k8sauditeks/README.md:40]: ok

File: plugins/k8sauditeks/README.md
Comment (Issif) [reply plugins/k8sauditeks/README.md:35]: I'll switch them to `open_parameters`. 
What do you mean by "URI-like encoding form"?

File: plugins/k8sauditeks/README.md
Comment (Issif) [reply plugins/k8sauditeks/README.md:61]: I'm off for next 2 weeks, we can wait til my return

File: plugins/k8sauditeks/falco.yaml
Comment (Issif) [reply plugins/k8sauditeks/falco.yaml:4]: :sweat_smile:

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:78]: Ok

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:114]: It's a string enforced by AWS for the logs.

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:128]: :+1:

File: plugins/k8sauditeks/Makefile
Comment (jasondellaluce) [reply plugins/k8sauditeks/Makefile:22]: Yeah that works for me!

File: plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go:78]: For the region, I don't think we should make it required, I don't think either we should set a default value. Most users will run Falco on EKS, and by design, the SDK will access to the correct endpoints of AWS API. The region is only used for that. wydt?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (geraldcombs) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:45]: Can you add a title for each of these as well?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (geraldcombs) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:51]: Can you add a title for each of these along with a "default=" annotation?

File: plugins/k8sauditeks/README.md
Comment (Issif) [reply plugins/k8sauditeks/README.md]: I replaced the ruleset by a single dummy rule to allow people to test the plugin, we could change that in future, wdyt?

Comment (jasondellaluce) [reply plugins/k8sauditeks/README.md]: I think the way to go for now would be to remove the `k8s_audit` requirement from `required_plugin_versions` of the actual k8saudit ruleset, so that users can test this against the real k8s audit rules and we don't have an unused dummy ruleset here.

Comment (Issif) [reply plugins/k8sauditeks/README.md]: I agree, but what can we do until that? I don't want to block this PR too long

File: plugins/k8saudit-eks/Makefile
Comment (jasondellaluce) [inline plugins/k8saudit-eks/Makefile:1]: The makefile of other plugins have been changed recently to support automatic readme generation. See https://github.com/falcosecurity/plugins/pull/138. I would be great if we could have those changes here as well.

For the future: maybe a makefile template (at least for Go plugins) might be the way to go.

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:160]: Leftover?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (jasondellaluce) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:172]: Leftover?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:160]: :sweat_smile:

File: plugins/k8saudit-eks/Makefile
Comment (Issif) [reply plugins/k8saudit-eks/Makefile:1]: :+1:

Comment (Issif) [reply plugins/k8saudit-eks/Makefile:1]: It's updated

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Kaizhe) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:70]: if `AWS_PROFILE` is not presented, should we look up for other environment variable as well?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Kaizhe) [inline plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:126]: aws supports two filter per log group which is not many, it's probably a good idea to support using the existing filter as an option.

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:70]: what env vars are you thinking about?

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:126]: I don't understand your point, the idea here is to create a nice UX, the users just set the name of their cluster and the filter is generated by the plugin.

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Kaizhe) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:70]: [AWS_WEB_IDENTITY_TOKEN_FILE](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)

Comment (Issif) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:70]: technically, it will work, because I'm using the official SDK, if the env var is there, it will be used

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Issif) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:126]: Ok, I didn't used a subscription, so we're not limited in number of filters per log group. Should be ok.

File: plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
Comment (Kaizhe) [reply plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:70]: sgtm

File: plugins/k8sauditeks/README.md
Comment (jasondellaluce) [reply plugins/k8sauditeks/README.md]: You should be all set with the new feature we introduced in Falco with https://github.com/falcosecurity/falco/pull/2190. It will be part of the upcoming Falco 0.33.0 release.

In your case, you should attain what you want by editing the official k8saudit ruleset with something like:
```yaml
- required_plugin_versions:
  - name: k8saudit
    version: 0.1.0
    alternatives:
      - name: k8saudit-eks
        version: 0.1.0
```

Comment (Issif) [reply plugins/k8sauditeks/README.md]: > You should be all set with the new feature we introduced in Falco with [falcosecurity/falco#2190](https://github.com/falcosecurity/falco/pull/2190). It will be part of the upcoming Falco 0.33.0 release.

I agree, it should work as is, I tried to respect

> In your case, you should attain what you want by editing the official k8saudit ruleset with something like:
> 
> ```yaml
> - required_engine_version: 15
> - required_plugin_versions:
>   - name: k8saudit
>     version: 0.1.0
>     alternatives:
>       - name: k8saudit-eks
>         version: 0.1.0
> ```

Ok, let me update the README to explain that

File: plugins/k8saudit-eks/rules/k8s_audit_rules.yaml
Comment (leogr) [inline plugins/k8saudit-eks/rules/k8s_audit_rules.yaml:31]: Could we remove the `rules/` folder from here?

(otherwise, the CI will automatically publish a package for that)

File: plugins/k8saudit-eks/go.mod
Comment (leogr) [inline plugins/k8saudit-eks/go.mod:7]: We should use a tagged version here.

**Review by leogr — COMMENTED** [review]
Left some comments.

We also need to add an entry into `registry.yaml`, but we can do it also in a follow-up PR.

File: plugins/k8saudit-eks/go.mod
Comment (Issif) [reply plugins/k8saudit-eks/go.mod]: I used the 0.3.0 :+1:

**Comment by Issif** [comment]
> Left some comments.
> 
> We also need to add an entry into `registry.yaml`, but we can do it also in a follow-up PR.

We agreed with @jasondellaluce to merge this merge first, then create a second one with the Changelog, the first tag and the update of the registry

**Comment by jasondellaluce** [comment]
> @leogr
We agreed with @jasondellaluce to merge this merge first, then create a second one with the Changelog, the first tag and the update of the registry

Of those, I think I would add the plugin to the registry right away even if not officially released/tagged, just with the purpose of assigning and reserving a plugin ID. If we want to wait for the first real release, I would suggest to open an issue after this PR gets merged to remind us that this plugin still needs to get an official ID assigned.

**Comment by Issif** [comment]
> > @leogr
> > We agreed with @jasondellaluce to merge this merge first, then create a second one with the Changelog, the first tag and the update of the registry
> 
> Of those, I think I would add the plugin to the registry right away even if not officially released/tagged, just with the purpose of assigning and reserving a plugin ID. If we want to wait for the first real release, I would suggest to open an issue after this PR gets merged to remind us that this plugin still needs to get an official ID assigned.

Ok, I reserved the ID with my last commit

**Comment by jasondellaluce** [comment]
> Ok, I reserved the ID with my last commit https://github.com/falcosecurity/plugins/pull/134/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R61
You should edit `registry.yaml`, not the readme (it is auto generated)

**Comment by Issif** [comment]
> > Ok, I reserved the ID with my last commit [#134 (files)](https://github.com/falcosecurity/plugins/pull/134/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R61)
> > You should edit `registry.yaml`, not the readme (it is auto generated)

Sorry, I did it, and changed the name to respect the naming convention enforces by the Makefile

File: plugins/k8saudit_eks/Makefile
Comment (leogr) [inline plugins/k8saudit_eks/Makefile:17]: I prefer k8saudit-eks. 

We prefer to use underscores only in source names. The dash is preferable for plugin names (I know, dummy_c is unfortunate 😅)

Comment (Issif) [reply plugins/k8saudit_eks/Makefile:17]: @leogr I prefer the dash too, but I followed the current convention, only the underscore is allowed, see:
https://github.com/falcosecurity/plugins/blob/master/build/registry/pkg/registry/validate.go#L24

```go
var (
	rgxName   = regexp.MustCompile(`^[a-z]+[a-z0-9-_]*$`)
	rgxSource = regexp.MustCompile(`^[a-z]+[a-z0-9_]*$`)
)
```

Comment (Issif) [reply plugins/k8saudit_eks/Makefile]: Ok, I know, I had to rebase on the upstream, in my locale version, the dash wasn't allows by the validate pkg, see:
```shell
❯ make check-registry
error: plugin name does follow the naming convention: 'k8saudit-eks'
```

After a rebase, it's ok.

Comment (leogr) [reply plugins/k8saudit_eks/Makefile]: :+1:

**Review by leogr — APPROVED** [review]

**Pull request merged**
